### ISSUE TITLE
Issue coupon discount with tax excluded and deliver free up to set value

### ISSUE THREAD

**Issue by barbosatrix** [PrestaShop/PrestaShop#21761]
**Describe the bug**

I created a $25000 pesos coupon for every user on the platform. When the user goes to pay, prestashop add some value to the coupon just if I set tax-excluded option on the Back office discount page.
I have free delivery when the order is up to $100000. If I let the coupon with tax included and the order is up to $100000, it not finish the chekout. If I let the coupon with tax excluded, prestashop finish the chekout but increase the coupon value in some way relate to producto tax 

**Expected behavior**
The expected behavior is finish the ckeckout with coupon and tax included parameter on the discount settings. 
In addition, not to add some value when I choose tax excluded on the coupon. 

**Steps to Reproduce**
**Steps to reproduce the behavior 1:**
1.	Create a $25000 coupon for any user 
2.	On coupon actions set tax excluded
3.	On Carrier set $0 to orders up to $100000
4.	From a user account make an order for more tan $100000 without count tax, goes to the checkout page and check the coupon value.

**Steps to reproduce the behavior 2:**
1.	Create a $25000 coupon for any user 
2.	On coupon actions set tax included
3.	On Carrier set $0 to orders up to $100000
4.	From a user account make an order for more tan $100000 tax excluded, and goes to the checkout page, select the carryer, try to finish the checkout. It brings you back to the home page without make checkout.
Screenshots
 
![Imagen1](https://user-images.githubusercontent.com/15943150/98181306-e6d9db00-1ed0-11eb-9c29-47bc6822516c.png)

![Imagen2](https://user-images.githubusercontent.com/15943150/98181336-f78a5100-1ed0-11eb-9ee1-79494929cfee.png)

![Imagen3](https://user-images.githubusercontent.com/15943150/98181345-fbb66e80-1ed0-11eb-975a-eca4085b5937.png)

![Imagen4](https://user-images.githubusercontent.com/15943150/98181353-ffe28c00-1ed0-11eb-8213-70749374e5b4.png)

 
**Additional information**
•	PrestaShop version: 1.7.6.5
•	PHP version: 7.2.28

**Comment by hibatallahAouadni** [comment]
Hello @barbosatrix 

I didn't manage to reproduce the issue with PS1.7.6.8 (please see screen record below)

https://drive.google.com/file/d/1IQfjQTmtjbIoSKdaIvXiQjYe35bjSemR/view

Please check and feedback.

Thanks!

**Comment by barbosatrix** [comment]
Hi @hibatallahAouadni  , than you for the help,
I was configured the free shipping from the carrier options like in the next picture but now I did it the way you did in the video on carrier preferences.
![Screenshot_1](https://user-images.githubusercontent.com/15943150/98250904-0a8b3880-1f46-11eb-951b-ec7a1e2432a8.png)

After that:

I tried to buy only one product and it works.
Then I tried to buy several products without taxes and it works.
Then I tried to buy products with taxes and it works.
Last I tested a cart mixing products with taxes and without taxes and it fails and sends me to the home page.
The test with the product mix should be done first because if it is tested last, after modifying a previous cart it works.

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @barbosatrix 

I didn't manage to reproduce the issue even with I put products with taxes and products without taxes in the same cart and proceed to checkout as you can see in the screen record below:

https://drive.google.com/file/d/1t3bgoC0Yt4i3Po3m05J9Q2pYTcCa9hfZ/view

Please check and feedback.

Thanks!

**Comment by barbosatrix** [comment]
It seems that is relate to payment method from mercadopago module because with another payment method it works fine.

**Comment by barbosatrix** [comment]
It seems that is relate to payment method from mercadopago module because with another payment method it works fine.
But im still have a question. Why if I use the tax excluded the price of the discount is increased?

**Comment by hibatallahAouadni** [comment]
Hello @barbosatrix 

So, your issue is not a PrestaShop core bug but a module problem.
We will close this issue.

PS: about your question 
> Why if I use the tax excluded the price of the discount is increased?

if You encounter a problem with this topic, could you please open a new issue.

Thanks!
